minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/ObservationFromDamageImplementation.java

```diff
@@ -0,0 +1,131 @@
+            this.damageSource = event.getSource();
+            this.damageAmount = event.getAmount();
+            this.entity = event.getEntityLiving();
+        }
+    }
+
+    private void resetDamage(){
+	    this.damageAmount = 0;
+	    this.damageSource = null;
+	    this.entity = null;
+	    this.hasDied = false;
+    }
+
+    @Override
+    public void writeObservationsToJSON(JsonObject json, MissionInit missionInit)
+    {
+        EntityPlayerSP player = Minecraft.getMinecraft().player;
+        JsonObject damage_json = new JsonObject();
+        json.addProperty("is_dead", player.isDead);
+        json.addProperty("living_death_event_fired", this.hasDied);
+        this.hasDied = false;
+        
+        if (this.damageAmount != 0 && this.damageSource != null) {
+            System.out.println(this.damageAmount + " damage from " + this.damageSource.getDamageType() + " by entity " + this.damageSource.getEntity());
+        }
+
+        if (this.damageAmount != 0) {
+            damage_json.addProperty("damage_amount", this.damageAmount);
+        }
+
+        if (this.damageSource != null) {
+            damage_json.addProperty("damage_type", this.damageSource.getDamageType());
+            damage_json.addProperty("hunger_damage", this.damageSource.getHungerDamage());
+            damage_json.addProperty("is_damage_absolute", this.damageSource.isDamageAbsolute());
+            damage_json.addProperty("is_fire_damage", this.damageSource.isFireDamage());
+            damage_json.addProperty("is_magic_damage", this.damageSource.isMagicDamage());
+            damage_json.addProperty("is_difficulty_scaled", this.damageSource.isDifficultyScaled());
+            damage_json.addProperty("is_explosion", this.damageSource.isExplosion());
+            damage_json.addProperty("is_projectile", this.damageSource.isProjectile());
+            damage_json.addProperty("is_unblockable", this.damageSource.isUnblockable());
+            damage_json.addProperty("death_message", this.damageSource.getDeathMessage(player).getUnformattedText());
+
+            if (this.damageSource.getEntity() != null) {
+                damage_json.addProperty("damage_entity", this.damageSource.getEntity().getName());
+                damage_json.addProperty("damage_entity_id", this.damageSource.getEntity().getEntityId());
+                JsonArray entity_equipment = new JsonArray();
+                // TODO do we need to mark the equipment slot of this armor?
+                for (ItemStack item : this.damageSource.getEntity().getEquipmentAndArmor()) {
+                    if (item.getItem().getRegistryName() != null) {
+                        JsonElement jitem = new JsonPrimitive(item.getItem().getRegistryName().toString());
+                        entity_equipment.add(jitem);
+                    }
+                }
+                damage_json.add("damage_entity_equipment", entity_equipment);
+            } if (this.damageSource.getDamageLocation() != null) {
```

Comment:
```suggestion
            }
            if (this.damageSource.getDamageLocation() != null) {
```

(since this isn't an else if)

Reply:
I think I got this with the reformatting